src/views/Correction.vue

```diff
@@ -127,6 +121,14 @@ export default class Correction extends Vue {
         throw new Error('Invalid correction type')
       }
 
+      // NB: specific entities are targeted via LaunchDarkly
+      if (!GetFeatureFlag('supported-correction-entities')?.includes(this.getEntityType)) {
+        window.alert('Corrections for this entity type are not available at the moment.\n' +
+          'Please check again later.')
+        this.$root.$emit('go-to-dashboard', true)
+        return
+      }
+
```

Comment:
What do you think about adding the LD check to VueRouter as a guard using the route meta data to set the flag like this:
```javascript
{
    path: '/correction',
    name: RouteNames.CORRECTION,
    component: Correction,
    meta: {
      launchDarklyFlag: 'supported-correction-entities',
      requiresAuth: true,
      isStaffOnly: true,
      filingType: FilingTypes.CORRECTION
    }
  }
  ```
This approach would avoid duplicating the launch darkly check in each view and it also simplifies our testing since the view can be tested without having to write workaround for the launch darkly check.

Reply:
That's a very good idea and I was already thinking how to implement it, but this page/route doesn't know the entity type until it loads the filing.

Unless we change that.

Reply:
If we fetched the business info earlier then we would already know the entity type, but there is no "earlier" since this is the route we start on.

I'll let thinking about this.

As for testing, it's actually easy to mock the feature flag call.

Reply:
Hmm, without putting logic in App.vue (to load the business info), I don't see how the router can do the FF check in this case. Do you have any ideas?

Reply:
Ah, good point.  If I understand correctly, launch darkly returns an array of entity types when given a launch darkly flag -- right?

Reply:
Yes, the FF is an array of allowable entity types (for the given filing, eg, correction).

We _could_ include entity type as an URL parameter. It seems a bit redundant since Edit UI can look it up but I do like the idea of the router being the "gatekeeper".

Note that we have different types of feature flags (eg, boolean, string, array), so the router would need to understand what each flag tests (ie, is specific to flag). What do you think about this?

Reply:
If you like the idea of router acting as a gatekeeper using FF, I suggest the cleanest way to implement this is to have the FF always return a Boolean.  In edit-ui, it looks like all other feature flags (besides the one in this PR) return Booleans.  Then it's easy to write a router guard that checks the appropriate FF.  For corrections, you could use the same pattern by adding the filing type to the flag like this:

- supported-correction-entities-BEN-enabled
- supported-correction-entities-SP-enabled
- supported-correction-entities-GP-enabled

Not sure, the wording above is the best, but I hope it illustrates the point.

Reply:
It illustrates the point, but ouch. We have many filing-array flags, and I think it's desirable to keep the number of flags to a minimum (but let me check with Patrick).

Do you want any changes in this PR?

Reply:
Patrick says we can add dozens of new FF without impact to our LaunchDarkly plan.

Reply:
Great discussion Sev.  I'll create a ticket to refactor the FF to use a VueRouter guard.